prime-router/metadata/fhir_mapping/hl7/segments/MSH/Provenance-Source.yml

```diff
@@ -0,0 +1,95 @@
+resourceType: Provenance
+id:
+  type: STRING
+  valueOf: "GeneralUtils.generateResourceId()"
+  expressionType: JEXL
+
+target:
+  valueOf: datatype/Reference
+  generateList: true
+  expressionType: resource
+  specs: $Bundle
+
+entity_1:
+  expressionType: nested
+  generateList: true
+  expressionsMap:
+    role:
+      type: STRING
+      valueOf: "source"
+
+# second entity value may be added after a mapping update based on MSH.3
+
+agent_1:
```

Comment:
Nit: I think we've been trying to collapse these into one schema element using `expressions`.  I think long term it forces keeping items like these close together so that we can verify the conditions remain mutually exclusive

Reply:
Ah right, I've been falling back onto whatever I was using as a starting point, but good call I'll rework this.